src/rust/engine/src/rule_graph.rs

```diff
@@ -790,8 +804,16 @@ impl RuleEdges {
     }
   }
 
-  fn add_edges_via(&mut self, selector_path: Vec<Selector>, new_dependencies: &Entries) {
-    if selector_path.is_empty() && !new_dependencies.is_empty() {
+  pub fn entries_for(&self, selector_path: &SelectorPath) -> Entries {
+    self.selector_to_dependencies.get(selector_path).map(|x|x.clone()).unwrap_or_else(|| Vec::new())
```

Comment:
`iter.map(|x|x.clone())` should be equivalent to `iter.cloned()` ('cloned' with a 'd')